### PR TITLE
Enabled Copy/Paste Feature in Text Fields

### LINKED ISSUE #470: Trying to use clipboard in menubar results in chaos.
**Describe the bug**
When you try to paste stuff in menubar's text input fields you accidentally end up pasting it on the simulator.  
**To Reproduce**

**Expected behavior**
Should paste stuff in the input text field and not in the menubar.

**Screenshots**

**Additional context**
The copy-pasting isn't working even if we don't have any simulator element in the clipboard.

**Are you working on this issue? (Yes/No)**
No

### PR DESCRIPTION
**Pull request by Shivansh2407**
Fixes #470 

#### Describe the changes you have made in this pr -
Here I have added a Javascript function to enable Copy/Paste feature. Before that whenever we used to copy the text in text field to clipboard then it used to copy the circuit element and not the text but now the text gets copied and nothing else. We can now copy and paste from the Project Name , Circuit Name and Combinational Analysis Popup Text fields.

### Screenshots of the changes (If any) -
patch22

### CODE REVIEW

**Comment by Shivansh2407** [comment]
@tachyons and @ankitkataria please review this 👍

**Comment by yash15601** [comment]
@Shivansh2407 good work😄

File: public/js/listeners.js [hunk 1]
```diff
@@ -52,7 +52,11 @@ function startListeners() {
     });
     window.addEventListener('mousemove', onMouseMove);
 
-
+    window.addEventListener('copy', (event) => {
```
Comment (satu0king) [inline public/js/listeners.js]: By doing this, the listeners that do the regular copy/paste of circuits `document.addEventListener('copy', function(e) ...` stopped working.

Comment (Shivansh2407) [reply public/js/listeners.js]: Will Fix this soon @satu0king Sir.

Comment (Shivansh2407) [reply public/js/listeners.js]: @satu0king can you please suggest what can be done here ?

Comment (satu0king) [reply public/js/listeners.js]: @yash15601 I am actually not sure myself. 

The issue is this 
1. The canvas needs to have a set of listeners to do copy/paste in canvas
1. The property menu text boxes should work as well and the canvas listeners shouldn't work then
1. Delete key on mac is `backspace` and that needs to be supported, unfortunately in the textbox, backspace will delete of the element.
1. Once a textbox is selected, and the canvas is clicked, the focus needs to change back to the canvas.

Comment (yash15601) [reply public/js/listeners.js]: @satu0king   4 is working properly will check other

Comment (Shivansh2407) [reply public/js/listeners.js]: > @yash15601 I am actually not sure myself.
> 
> The issue is this
> 
> 1. The canvas needs to have a set of listeners to do copy/paste in canvas
> 2. The property menu text boxes should work as well and the canvas listeners shouldn't work then
> 3. Delete key on mac is `backspace` and that needs to be supported, unfortunately in the textbox, backspace will delete of the element.
> 4. Once a textbox is selected, and the canvas is clicked, the focus needs to change back to the canvas.

Will fix these issues soon @satu0king Sir.

**Comment by satu0king** [comment]
Not applicable anymore as this is the old UI.  Thanks!

**Pull request closed without merging**